packages/git/src/browser/history/git-history-widget.ts

```diff
@@ -196,6 +203,10 @@ export class GitHistoryWidget extends GitBaseWidget<GitHistoryListNode> implemen
             expansionToggleIcon = "caret-down";
         }
         const headEl = [];
+        const hash = Md5.hashStr(commit.authorEmail);
+        const gravatar = h.div({ className: "image-container" },
+            h.img({ className: "gravatar", src: `https://www.gravatar.com/avatar/${hash}?d=robohash` }));
```

Comment:
Could you extract the computation of the avatar image to a service, that can be injected (and rebound)?